### PR TITLE
Change Ctrl+C behavior (plus minor unrelated fixes)

### PR DESCRIPTION
**Pull request by danielballan**
One Ctrl+C pauses (if resuming is possible) and gives this message.

```
In [4]: RE(checkpoint_forever())
^C
Your RunEngine is entering a paused state. These are your options for changing
the state of the RunEngine:

resume()  --> will resume the scan
 abort()  --> will kill the scan with an 'aborted' state to indicate
              the scan was interrupted
  stop()  --> will kill the scan with a 'finished' state to indicate
              the scan stopped normally

Pro Tip: Next time, if you want to abort, tap Ctrl+C twice quickly.

Out[4]: ['437eaf76-a6a8-439c-8e07-c877e245ce21']
```

When resuming is not possible, it aborts like this.

```
In [7]: RE(stepscan(det, motor))
^C
Pausing...
No checkpoint; cannot pause. Aborting...
Run aborted
Out[7]: ['e113059e-88d6-4ebb-943e-acec14369df2']
```

Double-tapping Ctrl+C aborts in one step.

```
In [8]: RE(checkpoint_forever())
^C^C
Aborting....
Run aborted
Out[8]: ['988620fd-eaa4-4897-ba2c-c723118dffc3']
```

Importantly, in all cases, once the prompt is returned the user, bluesky surrenders control of SIGINT (Ctrl+C) and lets it behave normally.

### CODE REVIEW

**Comment by danielballan** [comment]
@stuwilkins @cmazzoli @ambarb

File: bluesky/run_engine.py [hunk 1]
```diff
@@ -199,8 +200,8 @@ def __init__(self, md=None, logbook=None):
         self._bundling = False  # if we are in the middle of bundling readings
         self._run_is_open = False  # if we have emitted a RunStart, no RunStop
         self._deferred_pause_requested = False  # pause at next 'checkpoint'
-        self._sigint_handler = None
-        self._sigtstp_handler = None
+        self._sigint_handler = None  # intercepts Ctrl+C
+        self._sigtstp_handler = None  # intercepts Ctrl+Z
```
Comment (tacaswell) [inline bluesky/run_engine.py]: Don't need this

File: bluesky/run_engine.py [hunk 2]
```diff
@@ -656,9 +658,18 @@ def _check_for_signals(self):
...
         if self.state.is_running:
             if self._sigint_handler.interrupted:
-                self.debug("RunEngine detected a SIGINT (Ctrl+C)")
-                loop.call_soon(self.request_pause, False, 'SIGINT')
                 self._sigint_handler.interrupted = False
+                with SignalHandler(signal.SIGINT) as second_sigint_handler:
+                    self.debug("RunEngine detected a SIGINT (Ctrl+C)")
+                    # We know we will either pause or abort, so we can
+                    # hold up the scan now....
+                    ttime.sleep(0.5)  # give user time to hit Ctrl+C again
```
Comment (tacaswell) [inline bluesky/run_engine.py]: does this eat _all_ sigint for half a second?  Might be better to do this as a loop over a shorter interval

**Comment by danielballan** [comment]
@tacaswell changes addressed

**Pull request merged**